crates/resctrl/src/lib.rs

```diff
@@ -368,12 +502,127 @@ mod tests {
+        assert_eq!(info.mounted, false);
+        assert_eq!(info.mount_point, None);
+        assert_eq!(info.writable, false);
+    }
+
+    #[test]
+    fn test_detect_support_mounted_and_writable() {
+        let mut fs = MockFs::default();
+        // Simulate mounted under default root
+        fs.add_file(Path::new("/proc/mounts"), "resctrl /sys/fs/resctrl resctrl rw 0 0\n");
+        fs.add_dir(Path::new("/sys"));
+        fs.add_dir(Path::new("/sys/fs"));
+        fs.add_dir(Path::new("/sys/fs/resctrl"));
+        fs.add_file(Path::new("/sys/fs/resctrl/tasks"), "");
+        let rc = Resctrl::with_provider(fs, Config::default());
+        let info = rc.detect_support().expect("detect ok");
+        assert_eq!(info.mounted, true);
+        assert_eq!(info.mount_point, Some(PathBuf::from("/sys/fs/resctrl")));
+        assert_eq!(info.writable, true);
+    }
+
+    #[test]
+    fn test_detect_support_mounted_but_no_permission() {
+        let mut fs = MockFs::default();
+        fs.add_file(Path::new("/proc/mounts"), "resctrl /sys/fs/resctrl resctrl rw 0 0\n");
+        fs.add_dir(Path::new("/sys/fs/resctrl"));
+        let tasks = Path::new("/sys/fs/resctrl/tasks");
+        fs.add_file(tasks, "");
+        fs.set_no_perm_file(tasks);
+        let rc = Resctrl::with_provider(fs, Config::default());
+        let info = rc.detect_support().expect("detect ok");
+        assert_eq!(info.mounted, true);
+        assert_eq!(info.writable, false);
+    }
+
+    #[test]
+    fn test_ensure_mounted_respects_auto_mount_flag() {
+        let mut fs = MockFs::default();
+        fs.add_file(Path::new("/proc/mounts"), "");
+        let rc = Resctrl::with_provider(
+            fs,
+            Config { root: PathBuf::from("/sys/fs/resctrl"), group_prefix: "pod_".into(), auto_mount: false },
+        );
+        let err = rc.ensure_mounted().unwrap_err();
+        match err {
+            Error::NotMounted { .. } => {}
+            other => panic!("unexpected: {other:?}"),
+        }
+    }
+
+    #[test]
+    fn test_ensure_mounted_performs_mount() {
+        let mut fs = MockFs::default();
+        fs.add_file(Path::new("/proc/mounts"), "");
+        // also ensure /sys and /sys/fs exist in mock
+        fs.add_dir(Path::new("/sys"));
+        fs.add_dir(Path::new("/sys/fs"));
+        let rc = Resctrl::with_provider(
+            fs.clone(),
+            Config { root: PathBuf::from("/sys/fs/resctrl"), group_prefix: "pod_".into(), auto_mount: true },
+        );
+        rc.ensure_mounted().expect("mounted");
+        // After mount, detect reports mounted
+        let info = rc.detect_support().expect("detect ok");
+        assert!(info.mounted);
+        assert_eq!(info.mount_point.unwrap().to_string_lossy(), "/sys/fs/resctrl");
+    }
+
+    #[test]
+    fn test_ensure_mounted_permission_failure() {
+        let mut fs = MockFs::default();
+        fs.add_file(Path::new("/proc/mounts"), "");
```

Comment:
we might also test what happens when /proc/mounts does not exist or has a permission issue